src/io/aleph/dirigiste/Pool.java

```diff
@@ -311,8 +313,12 @@ private void startControlLoop(int duration, int iterations) {
                     _taskCompletionRates.sample(key, completed * _rateMultiplier);
                     _taskRejectionRates.sample(key, rejected * _rateMultiplier);
 
-                    double utilization = 1.0 - ((available - queueLength) / Math.max(1.0, objects));
+                    // To ensure we'll compute the correct utilization, we need to lock to prevent
+                    // objects creation.
+                    _lock.lock();
+                    double utilization = getUtilization(available, q.getQueueLength(), q.objects.get());
                     _utilizations.sample(key, utilization);
+                    _lock.unlock();
```

Comment:
We need to get the lock to compute the correct utilization, otherwise, we might get  `0`
while an object is currently on creation.